model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java

```diff
@@ -991,6 +992,8 @@ private List<Predicate> predicates(Map<String, String> attributes, Root<UserEnti
                 default:
                     Join<UserEntity, UserAttributeEntity> attributesJoin = root.join("attributes", JoinType.LEFT);
 
+                    if (value.length() > 255) throw new BadRequestException(String.format("Searching by attribute values longer that 255 characters is not supported by %s.", JpaUserProvider.class.getClass().getSimpleName()));
```

Comment:
Is it OK to throw JAX-RS exceptions from the storage layer? This one will also leak internal details (ok, just the class name but still :)).

Reply:
Agreed, I believe the correct way is to throw a `ModelException` or something similar and then the service layer should translate. On the other hand I am not sure what we are currently doing in the service layer, I remember we don't have a consistent approach to exceptions